lms/djangoapps/discussion_api/api.py

```diff
@@ -714,7 +730,7 @@ def delete_thread(request, thread_id):
 
     """
     cc_thread, context = _get_thread_and_context(request, thread_id)
-    if can_delete(cc_thread, context):
+    if has_permission(cc_thread, context):
```

Comment:
@nasthagiri 
For now, same check '_is_author_or_privileged(cc_content, context)' is being used for all three cases.
If I go for first option then there will be repetitive code for 'can_delete' and 'can_view' and for second option; the action parameter could be passed to has_permission but that parameter would not make any difference because all three actions would point to same line of code.